projects/entity-data-api/src/Mirza/EntityDataAPI/Main.hs

```diff
@@ -31,35 +38,86 @@ main = (decodeEnv :: IO (Either String Opts)) >>= \case
   Left err -> fail $ "Failed to parse Opts: " <> err
   Right opts -> do
     print opts
-    launchProxy opts
+    multiplexInitOptions opts
+
+multiplexInitOptions :: Opts -> IO ()
+multiplexInitOptions opts = do
+  ctx <- initContext opts
+  putStrLn $ "Initialized context. Starting app on mode " <> (show . appMode $ opts)
+  case appMode opts of
+    Proxy     -> launchProxy ctx
+    API       -> launchUserManager ctx
+    Bootstrap -> do
+      res <- tryAddBootstrapUser ctx
+      print res
+
+promptLine :: String -> IO String
+promptLine prompt = do
+  putStr prompt
+  getLine
+
+
+tryAddUser :: AuthContext -> IO (Either AppError Bool)
+tryAddUser ctx = do
+  (authorisedUserStr :: String) <- promptLine "Enter thy creds: "
+  (toAddUserStr :: String) <- promptLine "User you want to add: "
+  let (authorisedUserSub :: StringOrURI) = fromString authorisedUserStr
+  let (toAddUserSub :: StringOrURI) = fromString toAddUserStr
+  res <- runAppM ctx $ addUserSub authorisedUserSub toAddUserSub
+  case res of
+    Right True  -> putStrLn "Successfully added user"
+    Right False -> putStrLn "Failed to add the user."
```

Comment:
`False` actually means user already exists? Since this is being printed to stdout I see no reason to say why it failed.